runtime/browser/runtime.cc

```diff
@@ -192,6 +195,15 @@ void Runtime::CloseContents(content::WebContents* source) {
     observer_->OnRuntimeClosed(this);
 }
 
+void Runtime::ExitApplication() {
+  auto app_service =
+      XWalkRunner::GetInstance()->app_system()->application_service();
+  int rph_id = web_contents()->GetRenderProcessHost()->GetID();
+  application::Application* app =
+      app_service->GetApplicationByRenderHostID(rph_id);
+  app->Terminate();
```

Comment:
would  be better just `observer_->OnApplicationExitRequested(this);` // `Application` is implementing `Runtime::Observer` interface, so further:
```
OnApplicationExitRequested(Runtime*  runtime)  {
  Terminate();
}
```